### PR TITLE
build01: add telegraf profile

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: profiles/telegraf.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          disk.tagdrop = {
+            fstype = [ "tmpfs" "ramfs" "devtmpfs" "devfs" "iso9660" "overlay" "aufs" "squashfs" ];
+            device = [ "rpc_pipefs" "lxcfs" "nsfs" "borgfs" ];
+          };
+        };
+        outputs = {
+          prometheus_client = {
+            listen = ":9273";
+            metric_version = 2;
+            # TODO http auth
```
Comment (Mic92) [inline profiles/telegraf.nix:63]: I could start by scraping this with my existing prometheus setup - I have IRC alerts set up there. We can also deploy a Prometheus service later to both servers.

File: profiles/telegraf.nix [hunk 2]
```diff
@@ -0,0 +1,84 @@
+{ pkgs, lib, config, ... }: let
+  isVM = lib.any (mod: mod == "xen-blkfront" || mod == "virtio_console") config.boot.initrd.kernelModules;
+  secrets = import ../secrets.nix;
+in {
+  config = {
+    systemd.services.telegraf.path = [ pkgs.nvme-cli ];
+
+    # TODO http auth
+    #deployment.keys."telegraf-secret-env".text = secrets."telegraf-secret-env";
```
Comment (Mic92) [inline profiles/telegraf.nix:9]: You seem to add deployment keys to the same file. Is this the preferred way?

File: profiles/telegraf.nix [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        options = [ "NOPASSWD" ];
+      }];
+    }];
+    # avoid logging sudo use
+    security.sudo.configFile = ''
+      Defaults:telegraf !syslog,!pam_session
+    '';
+    # create dummy file to avoid telegraf errors
+    systemd.tmpfiles.rules = [
+      "f /var/log/telegraf/dummy 0444 root root - -"
```
Comment (Mic92) [inline profiles/telegraf.nix:81]: This is used for healtchecks. Timers can write updates to this file and prometheus will check if timers run successfully or at all.

File: profiles/telegraf.nix (hunk 1 above)
Comment (zimbatm) [reply profiles/telegraf.nix:63]: Is it necessary to hide the service behind auth? I don't think it's exporting anything sensitive.

**Review by zimbatm — APPROVED** [review]
sounds good on principle. welcome!

File: profiles/telegraf.nix (hunk 2 above)
Comment (ryantm) [reply profiles/telegraf.nix:9]: We have all the deployment.keys in one place, so we can avoid importing secrets.nix in multiple places.

File: profiles/telegraf.nix (hunk 1 above)
Comment (Mic92) [reply profiles/telegraf.nix:63]: Not really.

File: profiles/telegraf.nix (hunk 2 above)
Comment (Mic92) [reply profiles/telegraf.nix]: dropped from the profile.

File: profiles/telegraf/default.nix [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                    while ("${pkgs.zfs}/bin/zpool status" | getline) {
+                        if ($1 ~ /pool:/) { printf "zpool_status,name=%s ", $2 }
+                        if ($1 ~ /state:/) { printf " state=\"%s\",", $2 }
+                        if ($1 ~ /errors:/) {
+                            if (index($2, "No")) printf "errors=0i\n"; else printf "errors=%di\n", $2
+                        }
+                    }
+                }
+              '')
+              (pkgs.runCommandNoCC "mdraid-health" {
```
Comment (Mic92) [inline profiles/telegraf/default.nix:45]: Added this one as well. Not fully tested yet since I don't run software raids.

**Pull request merged**